### PR TITLE
Add New Receive Buffer Unit Tests

### PR DESCRIPTION
**Pull request by nibanks**
## Description

We will be refactoring `QUIC_RECV_BUFFER` to support multiple parallel outstanding reads, so let's add some tests first to make sure we don't regress things with the upcoming refactor.

## Testing

Added

## Documentation

N/A - internal interface

### CODE REVIEW

File: src/core/unittest/RecvBufferTest.cpp [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+        return QuicRecvBufferInitialize(&RecvBuf, AllocBufferLength, VirtualBufferLength, CopyOnDrain ? TRUE : FALSE, PreallocChunk);
+    }
+    uint64_t GetTotalLength() {
+        return QuicRecvBufferGetTotalLength(&RecvBuf);
+    }
+    void SetVirtualBufferLength(uint32_t Length) {
+        QuicRecvBufferSetVirtualBufferLength(&RecvBuf, Length);
+    }
+    bool HasUnreadData() {
+        return QuicRecvBufferHasUnreadData(&RecvBuf) != FALSE;
```
Comment (csujedihy) [inline src/core/unittest/RecvBufferTest.cpp:50]: nit: no need to have "!= FALSE"?

Comment (nibanks) [reply src/core/unittest/RecvBufferTest.cpp:50]: It's converting between `BOOLEAN` and `bool`.

File: src/core/unittest/RecvBufferTest.cpp [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    ASSERT_EQ(
+        QUIC_STATUS_SUCCESS,
+        RecvBuf.Write(
+            WriteOffset,
+            WriteLength,
+            &InOutWriteLength,
+            &ReadyToRead));
+    ASSERT_EQ(WriteOffset == 0, ReadyToRead != FALSE); // Only ready to read if we wrote to the front
+    ASSERT_EQ(WriteLength+WriteOffset, InOutWriteLength); // All data was newly written
+    ASSERT_EQ(WriteLength+WriteOffset, RecvBuf.GetTotalLength()); // otal length should be offset plus write length
```
Comment (csujedihy) [inline src/core/unittest/RecvBufferTest.cpp:126]: *total

**Comment by ProjectsByJackHe** [comment]
LGTM! All BVT tests are passing. Some small spelling mistakes, but other than that, more unit testing for internal interfaces is good.

File: src/core/unittest/RecvBufferTest.cpp (hunk 2 above)
Comment (csujedihy) [reply src/core/unittest/RecvBufferTest.cpp:126]: also nit: WriteLength+WriteOffset => WriteLength + WriteOffset

**Review by csujedihy — APPROVED** [review]

**Pull request merged**